python/jiminy_py/src/jiminy_py/viewer.py

```diff
@@ -507,103 +700,168 @@ def _updateGeometryPlacements(self, visual=False):
 
-        R_pnc = rpyToMatrix(np.array(rotation))
+        # Perform the desired rotation
         if Viewer.backend == 'gepetto-gui':
-            H_abs = SE3(R_pnc, np.array(translation))
-            if relative:
-                H_orig = XYZQUATToSe3(self._client.getCameraTransform(self._window_id))
+            H_abs = SE3(rpyToMatrix(np.asarray(rotation)), np.asarray(translation))
+            if relative is None:
+                self._client.setCameraTransform(
+                    self._window_id, se3ToXYZQUAT(H_abs).tolist())
+            else:
+                # Not using recursive call for efficiency
                 H_abs = H_abs * H_orig
-            self._client.setCameraTransform(self._window_id, se3ToXYZQUAT(H_abs).tolist())
-        else:
-            if relative:
-                raise RuntimeError("'relative'=True not available with meshcat.")
-            import meshcat.transformations as tf
-            # Transformation of the camera object
-            T_meshcat = tf.translation_matrix(translation)
-            self._client.viewer["/Cameras/default/rotated/<object>"].set_transform(T_meshcat)
-            # Orientation of the camera object
-            Q_pnc = Quaternion(R_pnc).coeffs()
-            Q_meshcat = np.roll(Q_pnc, shift=1)
-            R_meshcat = tf.quaternion_matrix(Q_meshcat)
-            self._client.viewer["/Cameras/default"].set_transform(R_meshcat)
-
-    def captureFrame(self):
+                self._client.setCameraTransform(
+                    self._window_id, se3ToXYZQUAT(H_abs).tolist())
+        elif Viewer.backend == 'meshcat':
+            if relative is None:
+                # Meshcat camera is rotated by -pi/2 along Roll axis wrt the usual convention in robotics
+                translation = CAMERA_INV_TRANSFORM_MESHCAT @ translation
+                rotation = matrixToRpy(CAMERA_INV_TRANSFORM_MESHCAT @ rpyToMatrix(rotation))
+                self._client.viewer["/Cameras/default/rotated/<object>"].set_transform(
+                    mtf.compose_matrix(translate=translation, angles=rotation))
+            else:
+                H_abs = SE3(rpyToMatrix(np.asarray(rotation)), np.asarray(translation))
+                H_abs = H_abs * H_orig
+                self.set_camera_transform(H_abs.translation, rotation)  # The original rotation is not modified
+
+    def capture_frame(self, width=None, height=None, raw_data=False):
         if Viewer.backend == 'gepetto-gui':
-            png_path = next(tempfile._get_candidate_names())
-            self._client.captureFrame(self._window_id, png_path)
-            rgb_array = np.array(Image.open(png_path))[:, :, :-1]
+            if width is not None or height is None:
+                logging.warning("Cannot specify window size using gepetto-gui.")
+            assert not raw_data, "Raw data mode is not available using gepetto-gui."
+            png_path = next(tempfile._get_candidate_names()) + ".png"
```

Comment:
add comment  `# Gepetto is not able to save the image if it does not have .png extension`